packages/edit-navigation/src/components/layout/use-navigation-editor.js

```diff
@@ -8,11 +8,17 @@ import { useState, useEffect } from '@wordpress/element';
  * Internal dependencies
  */
 import { store as editNavigationStore } from '../../store';
+import { store as noticesStore } from '@wordpress/notices';
+import { __ } from '@wordpress/i18n';
 
 export default function useNavigationEditor() {
 	const [ hasFinishedInitialLoad, setHasFinishedInitialLoad ] = useState(
 		false
 	);
+	const [ selectedMenuId, setSelectedMenuId ] = useState( null );
+	const [ isMenuDeleted, setIsMenuDeleted ] = useState( false );
+	const [ isMenuBeingDeleted, setIsMenuBeingDeleted ] = useState( false );
```

Comment:
There is a `isDeletingEntityRecord` selector that should be able to serve the same purpose.